src/arglist.c

```diff
@@ -697,13 +698,18 @@ do_argfile(exarg_T *eap, int argn)
 	return;
     }
 
+    if (!is_split_cmd
+	    && (&ARGLIST[argn])->ae_fnum != curbuf->b_fnum
+	    && !check_can_set_curbuf_forceit(eap->forceit))
+	return;
```

Comment:
I think `Test_rewind` already does this, no? `s:make_args_list()` creates an argslist, calls `set winfixbuf`, then I call `next!` to force it to the next element of the args list, then call `rewind` and it fails with `E1513` but allows `rewind!`

Reply:
okay fine then.

Reply:
I follow these steps to make sure a test exercises a line in the code:

   cd src/testdir
   VIMRUNTIME=../../runtime gdb ../vim
   <add a breakpoint at the line which should be exercised>
   run -f  -u unix.vim --gui-dialog-file guidialog -U NONE --noplugin --not-a-term -S runtest.vim test_normal.vim

This will run the test_normal.vim test.  If a line is covered by the tests, then you will hit the breakpoint.  Otherwise you need to analyze the test.  In this PR, some lines are not exercised by the tests.

Reply:
Interesting, alright I'll need to remember that snippet. Maybe it'd be worth putting that in the codecov report message somehow? So people can see how to double-check

Reply:
This PR needs quite a few additional tests.

Reply:
It does? I did mention in several replies that some could be missing but I at least tried cover every mapping and command that could change a window's buffer. Admittedly I didn't touch all mappings / commands that should not be affected by `'winfixbuf'` (because they make a split or some other reason). What tests are missing?

Reply:
I was going by the Vim code coverage report (https://app.codecov.io/gh/vim/vim).  But it looks like
this report generation is broken (https://github.com/vim/vim/pull/14127).
I see in my local setup that these lines are indeed exercised by the tests.